comet/broker.py

```diff
@@ -146,28 +86,24 @@ def float_to_datetime(fl):
 
 @app.route('/register-state', methods=['POST'])
 async def registerState(request):
-    """Register a dataset state with CoMeT (the broker).
+    """Register a dataset state with the comet broker.
 
     This should only ever be called by kotekan's datasetManager.
     """
-    global states
-    global requested_states
-
     hash = request.json['hash']
     logger.debug('register-state: Received register state request, hash: {}'.format(hash))
     reply = dict(result="success")
 
-    async with lock_states:
-        state = states.get(hash)
+    # Lock states and check if the received state is already known.
+    async with await lock_manager.lock("states_lock"):
+        state = await redis.execute("hget", "states", hash)
```

Comment:
This is in the locked region so that no two processes can find that state to be missing and adding it to the requested states.
Wouldn't moving python code into a lua redis script just move the blocking time from waiting for the lock to waiting for the global redis lock?
Anyways, I think you are right an it will probably be more performant scripting this in redis just for having less requests sent to the redis server.